src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/CollectTraceTests.cs

```diff
@@ -82,6 +84,49 @@ public Task StopOnEvent_DoesNotStop_WhenPayloadFieldValueMismatch()
             });
         }
 
+        [Fact]
+        public async Task ExitOnStdinDisconnect_Succeeds()
+        {
+            // Start an app in 'connect' mode to match what the Visual Studio profiler does.
+            DiagnosticPortHelper.Generate(
+                DiagnosticPortConnectionMode.Listen,
+                out DiagnosticPortConnectionMode appConnectionMode,
+                out string diagnosticPortPath);
+
+            await using AppRunner appRunner = new(_outputHelper, Assembly.GetExecutingAssembly());
+            appRunner.ConnectionMode = appConnectionMode;
+            appRunner.DiagnosticPortPath = diagnosticPortPath;
+            appRunner.ScenarioName = TestAppScenarios.AspNet.Name;
+            using (CancellationTokenSource appStartCancellationTokenSource = new(CommonTestTimeouts.StartProcess))
+            {
+                // Start the app, but don't wait for it to be ready since we're not going to make any requests.
+                await appRunner.StartAsync(appStartCancellationTokenSource.Token, waitForReady: false);
+            }
+
+            // Start dotnet-monitor in `--exit-on-stdin-disconnect` mode.
+            await using MonitorCollectRunner toolRunner = new(_outputHelper);
+            toolRunner.ConnectionModeViaCommandLine = DiagnosticPortConnectionMode.Listen;
+            toolRunner.DiagnosticPortPath = diagnosticPortPath;
+            toolRunner.DisableAuthentication = true;
+            toolRunner.ExitOnStdinDisconnect = true;
+            await toolRunner.StartAsync();
+            Assert.False(toolRunner.HasExited);
+
+            // Close the tool's stdin, which should cause it to exit.
+            toolRunner.StandardInput.Close();
+
+            using (CancellationTokenSource monitorExitCancellationTokenSource = new(TimeSpan.FromSeconds(10)))
```

Comment:
nit: can you use or add something to src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/TestTimeouts.cs? The test environment can be very slow here so sometimes we have to up wait times.